File: js/recursionChallenge.js
exports.factorial = function(x) {
    if (x === 1){
        return 1
    } else {
        return (x * factorial(x-1))
    }

};

exports.palindrome = function(word) {
        originalWord = word.toString().toLowerCase().replace(/[^a-zA-z0-9]/g, "").split('')//.join('')
    
        if (originalWord.length == 1) {
            return true
        } else {
            let left = originalWord.shift()
            // console.log(left)
            let right = originalWord.pop()
            // console.log(right)
            if (left  == right) {
                return palindrome(originalWord.join(''))
            }
        } return false
};

exports.bottles = function() {

};

exports.toRoman = (num) => {
    const map = new Map()
    map.set('M', 1000);
    map.set('CM', 900);
    map.set('D', 500)
    map.set('CD', 400)
    map.set('C', 100)
    map.set('XC', 90)
    map.set('L', 50)
    map.set('XL', 40)
    map.set('X', 10)
    map.set('IX', 9)
    map.set('V', 5)
    map.set('IV', 4)
    map.set('I', 1)
    if(num === 0){
        return ''
    } else {
        for(let [key, value] of map){
            while(num >= value){
                return key + toRoman(num - value)
            }  
        }
    }
};

File: python/recursion_challenge.py
import re
def factorial(x):
	if x == 1:
		return 1
	return x * factorial(x-1)

def palindrome(word):
	clean_word = (re.sub('[^a-zA-Z0-9]','',word))
	original_word = list(clean_word.lower())
	
	if len(original_word) == 1:
		return True
	else:
		if original_word[0] == original_word[-1]:
			return palindrome("".join(original_word[1:-1]))
	return False

def to_roman(num):
	roman_to_arabic = {
        "I":1,
        "IV":4,
        "V":5,
        "IX":9,
        "X":10,
        "XL":40,
        "L":50,
        "XC":90,
        "C":100,
        "CD":400,
        "D":500,
        "CM":900,
        "M":1000,
    }

	if num == 0:
		return ''
	else:
		for key,value in reversed(roman_to_arabic.items()):
			while num >= value:
				return key + to_roman(num-int(value))


